clients/azuredevopsrepo/commits.go

```diff
@@ -205,18 +215,102 @@ func (handler *commitsHandler) listPullRequests() (map[string]clients.PullReques
 			Author: clients.User{
 				Login: *azdoPullRequest.CreatedBy.DisplayName,
 			},
-			HeadSHA:  *azdoPullRequest.LastMergeSourceCommit.CommitId,
+			HeadSHA:  *azdoPullRequest.LastMergeCommit.CommitId,
 			MergedAt: azdoPullRequest.ClosedDate.Time,
 		}
 	}
 
 	return pullRequests, nil
 }
 
-func (handler *commitsHandler) getFirstCommitCreatedAt() (time.Time, error) {
-	if err := handler.setup(); err != nil {
+func (c *commitsHandler) getFirstCommitCreatedAt() (time.Time, error) {
+	if err := c.setup(); err != nil {
 		return time.Time{}, fmt.Errorf("error during commitsHandler.setup: %w", err)
 	}
 
-	return handler.firstCommitCreatedAt, nil
+	return c.firstCommitCreatedAt, nil
+}
+
+func (c *commitsHandler) listStatuses(ref string) ([]clients.Status, error) {
+	matched, err := regexp.MatchString("^[0-9a-f]{5,40}$", ref)
```

Comment:
what about `A-F`? also most of our other regexes are declared in the form of package vars. 

In this other example, it expects a full 40, but doesn't necessarily have to be.
```
var gitCommitHashRegex = regexp.MustCompile(`^[a-fA-F0-9]{40}$`)
```

Reply:
I didn't realize regex were case-sensitive by default in Go.

As for package vars versus local vars, is there any reason why one over the other if this is the only use?

Reply:
> As for package vars versus local vars, is there any reason why one over the other if this is the only use?

Mainly about when it gets compiled. If it's a package var, it's one time when the package is imported. If it's local to the function, it only gets compiled when used, but every time the function is called.

We can keep it local